core/shared/platform/common/posix/posix_thread.c

```diff
@@ -166,17 +166,27 @@ int os_cond_wait(korp_cond *cond, korp_mutex *mutex)
 static void msec_nsec_to_abstime(struct timespec *ts, uint64 usec)
 {
     struct timeval tv;
-    long int tv_sec_new, tv_nsec_new;
+    time_t tv_sec_new;
+    long int tv_nsec_new;
+
+    /*
+     * Assumptions:
+     *  - time_t has no padding
+     *  - time_t is a signed integer
+     */
+
+    const time_t TIME_T_MAX =
+        (((time_t)1 << ((sizeof(time_t) * CHAR_BIT) - 2)) - 1) * 2 + 1;
```

Comment:
This code is hard to understand, is there any links or documentations to illustrate such kind of usage? I'm not sure if this is suitable for every platform.

If this is only to fix build issues on some specified platforms, I'd suggest you using platform defined macros.

Reply:
@xujuntwt95329 
can you tell me which headers are appropriate for such macros? i'm not familiar with this codebase.
core/shared/platform/nuttx/platform_internal.h etc?

Reply:
The macro `BH_PLATFORM_XXX` should be available here.

But still, I'm worried about this statement `(((time_t)1 << ((sizeof(time_t) * CHAR_BIT) - 2)) - 1) * 2 + 1`, nobody will know what this exactly mean after a couple of days. Is this the only approach to fix your build?

Reply:
it's just the easiest way it hopped up in my mind.

alternatively, i guess we can have for nuttx
```
#define BH_TIME_T_MAX 0xffffffff
```
and for others
```
#define BH_TIME_T_MAX LONG_MAX
```
how do you think?

Reply:
Looks good to me

Reply:
done